File: 1-two-sum/two-sum.py
# Problem: https://leetcode.cn/problems/two-sum/
# Accepted at: 2026年8月30日 10:51

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}

        for i in range(len(nums)):
            cur = nums[i]
            complement = target - cur
            if complement in map and map[complement] != i:
                return [map[complement],i]
            else:
                map[cur] = i
        
        return []

File: 1-two-sum/two-sum.ts
// Problem: https://leetcode.cn/problems/two-sum/
// Accepted at: 2026年6月29日 18:49

function twoSum(nums: number[], target: number): number[] {
    const indexRecord = new Map<number, number>;
    for (let i = 0; i < nums.length; i++) {
        let curNum = nums[i];
        let curTargetNum = target - curNum;
        let curTargetNumIndex = indexRecord.get(curTargetNum);
        if (curTargetNumIndex === undefined) {
            indexRecord.set(curNum,i);
            continue;
        }
        return [i,curTargetNumIndex]
    }
};
